python-checks/src/main/java/org/sonar/python/checks/SklearnPipelineSpecifyMemoryArgumentCheck.java

```diff
@@ -72,26 +76,33 @@ private static void createIssue(SubscriptionContext subscriptionContext, CallExp
     issue.addQuickFix(quickFix);
   }
 
-  private static boolean isPipelineCreation(CallExpression callExpression) {
-    return Optional.ofNullable(callExpression.calleeSymbol())
-      .map(Symbol::fullyQualifiedName)
-      .map(fqn -> "sklearn.pipeline.Pipeline".equals(fqn) || "sklearn.pipeline.make_pipeline".equals(fqn))
-      .orElse(false);
-  }
-
   private static boolean isUsedInAnotherPipeline(Name name) {
     Symbol symbol = name.symbol();
     return symbol != null && symbol.usages().stream().filter(usage -> !usage.isBindingUsage()).anyMatch(u -> {
       Tree tree = u.tree();
       CallExpression callExpression = (CallExpression) TreeUtils.firstAncestorOfKind(tree, Tree.Kind.CALL_EXPR);
       while (callExpression != null) {
-        Optional<String> fullyQualifiedName = Optional.ofNullable(callExpression.calleeSymbol()).map(Symbol::fullyQualifiedName);
-        if (fullyQualifiedName.isPresent() && isPipelineCreation(callExpression)) {
+        if (isPipelineCreation(callExpression) || isUsedBySklearnComposeEstimator(callExpression)) {
           return true;
         }
         callExpression = (CallExpression) TreeUtils.firstAncestorOfKind(callExpression, Tree.Kind.CALL_EXPR);
       }
       return false;
     });
   }
+
+  private static boolean isPipelineCreation(CallExpression callExpression) {
+    return Optional.ofNullable(callExpression.calleeSymbol())
+      .map(Symbol::fullyQualifiedName)
```

Comment:
This code looks quite good but we are checking twice if the symbol is null and if the qualified name is null as well. We could reactor this, and just directly check the FQN against `sklearn.compose`  and `Pipeline`,  etc...